### ISSUE TITLE
[Bug]:  Regression with recompile times when upgrading from  0.6.0 to 0.6.2 (~20% increase)

### ISSUE THREAD

**Issue by pushkargupta9807** [web-infra-dev/rspack#6304]
### System Info

  System:
    OS: Linux 6.2 Ubuntu 20.04.6 LTS (Focal Fossa)
    CPU: (32) x64 AMD EPYC 7763 64-Core Processor
    Memory: 113.67 GB / 125.78 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 20.11.0 - /usr/local/share/nvm/versions/node/v20.11.0/bin/node
    Yarn: 1.23.19 - /usr/bin/yarn
    npm: 10.2.4 - /usr/local/share/nvm/versions/node/v20.11.0/bin/npm
    Watchman: 20240317.093234.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 124.0.6367.60
  npmPackages:
    @rspack/cli: 0.6.2 => 0.6.2 
    @rspack/core: 0.6.2 => 0.6.2 

### Details

Hello, I just upgraded to 0.6.2 version, however I am seeing this slight regression with the recompile times. 

After the app is booted ,  a typical recompile/reload for a simple GUI change was taking ~5.5 seconds with 0.6.0 and now it takes around 7 seconds with 0.6.2 => 20% regression.  

### Reproduce Steps

Please refer to the details section - any simple GUI change on an already booted/ served app  should do it.

**Comment by SyMind** [comment]
@pushkargupta9807 I need the tracing file of your project. Please record it and provide it to me following the method described here: https://web-infra-dev.github.io/rspack-dev-guide/profiling/intro.html#chrome

**Comment by pushkargupta9807** [comment]
@SyMind I tried enabling the tracing file with the "rspack serve" command. But I did not find the trace file getting generated.  Does the trace file get generated with rspack serve command ?   I want for this case because the regression I am observing is related to recompile.

**Comment by pushkargupta9807** [comment]
Also there is definitely a regression - I just switched back to 0.6.0 and the recompile times are back to 5.5 seconds from 7 seconds for the same scenario. Am curious, 
1) have the rspack perf tests reported the same regression from 0.6.0 ?
2) is anyone else observing the same issue?

**Comment by SyMind** [comment]
@pushkargupta9807 Try run `RSPACK_PROFILE=TRACE=layer=chrome rspack serve`.
Afterward, make a modification to any file to trigger Hot Module Replacement (HMR), and then terminate the process.

**Comment by pushkargupta9807** [comment]
@SyMind I exactly did that - and then at the terminated the session using ctrl+c , but did not see the trace file getting generated . Let me try again today. Thanks!

**Comment by pushkargupta9807** [comment]
@SyMind in the meanwhile, I had these questions for you :

1. have the rspack perf tests reported the same regression from 0.6.0 ?
2. is anyone else observing the same issue?
3. could be there some config setting whose default value got changed from 0.6.0 ?

**Comment by SyMind** [comment]
@pushkargupta9807 no
